catalog/MDCCatalog/MDCCatalogStyle.m

```diff
@@ -16,10 +16,23 @@
 
 #import "MDCCatalogStyle.h"
 
+static inline BOOL isRunningiOS9OrGreater() {
```

Comment:
Looking at this more carefully, and having worked on some iOS 11 code, I think we should protect against older SDKs even on Xcode 7+.

```objectivec
static inline BOOL isRunningiOS9OrGreater() {
#if defined(__IPHONE_9_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_9_0)
   static NSOperatingSystemVersion iOS9Version = {9, 0, 0};
   NSProcessInfo *processInfo = [NSProcessInfo processInfo];
   return [processInfo isOperatingSystemAtLeastVersion:iOS9Version];
#endif
  return NO;
}
```

Does that make sense?
1. Check that iOS 9 is defined (Xcode supports iOS 9+)
2. Check that the SDK supported version includes iOS 9 as a minimum (base SDK supports iOS 9+)
3. Then finally use the runtime check to verify that we're actually running on iOS 9+ (won't execute on iOS 8)

Reply:
Yeah, it makes sense it just adds another check to see if the current version of XCode that the user is on supports newer versions of iOS, right?

Reply:
It's really for extra safety and it's a combination of two checks (if I understand correctly):

```
#if defined(__IPHONE_9_0)
```
This checks that iOS 9 is defined in the build environment, to make sure we don't try to use a symbol (like a selector or constant) that isn't defined in earlier SDKs.

```
&& (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_9_0)
```
This checks that the base SDK being used actually supports iOS 9.0 or greater. For example, someone could use the iOS 7 SDK on Xcode 7 even though Xcode 7 defaults to iOS 8.